src/gtk/window.cpp

```diff
@@ -3371,17 +3378,84 @@ wxEmitPressAndTapEvent(GdkEventTouch* gdk_event, wxWindow* win)
     win->GTKProcessEvent(event);
 }
 
+static void wxEventButtonFromEventTouch(GdkEventButton* gdk_event_button, const GdkEventTouch* gdk_event_touch) {
+    gdk_event_button->type = GDK_BUTTON_PRESS;
+    gdk_event_button->window = gdk_event_touch->window;
+    gdk_event_button->send_event = gdk_event_touch->send_event;
+    gdk_event_button->time = gdk_event_touch->time;
+    gdk_event_button->x = gdk_event_touch->x;
+    gdk_event_button->y = gdk_event_touch->y;
+    gdk_event_button->axes = gdk_event_touch->axes;
+    gdk_event_button->state = gdk_event_touch->state;
+    gdk_event_button->device = gdk_event_touch->device;
+    gdk_event_button->x_root = gdk_event_touch->x_root;
+    gdk_event_button->y_root = gdk_event_touch->y_root;
+
+    gdk_event_button->button = 1; // left button
+}
+
+static void wxEventMotionFromEventTouch(GdkEventMotion* gdk_event_motion, const GdkEventTouch* gdk_event_touch) {
+    gdk_event_motion->type = GDK_MOTION_NOTIFY;
+    gdk_event_motion->window = gdk_event_touch->window;
+    gdk_event_motion->send_event = gdk_event_touch->send_event;
+    gdk_event_motion->time = gdk_event_touch->time;
+    gdk_event_motion->x = gdk_event_touch->x;
+    gdk_event_motion->y = gdk_event_touch->y;
+    gdk_event_motion->axes = gdk_event_touch->axes;
+    gdk_event_motion->state = gdk_event_touch->state;
+    gdk_event_motion->device = gdk_event_touch->device;
+    gdk_event_motion->x_root = gdk_event_touch->x_root;
+    gdk_event_motion->y_root = gdk_event_touch->y_root;
```

Comment:
It looks like this is almost identical to the function above, so it would be IMO be better to have a single helper function doing this and taking the event type an argument.

Reply:
That is something I have been asking myself, indeed. But if you have a look at gdkevents.h,
you can that ```struct _GdkEventButton``` and ```struct _GdkEventMotion``` have basically the
same duplicated fields, without sharing a common aggregated structure or union.

So factorising the code means to arbitrarily chose one struct or the other for accessing the fields, 
(since there is no abstract 'GdkEventMouse' event struct),
and perform a hard cast that would not reveal problems at compilation time if the structure(s) change(s)
(the worst case being an inserted field, obviously) in gtk, but that would break at runtime. 
Does this sound acceptable ?

Reply:
Is there any problem with making the helper a template function?

Reply:
Not at all, I just fixed it as you suggest